src/main.py

```diff
@@ -46,9 +62,38 @@ def compute_means(model: AutoModelForCausalLM, tokenizer: AutoTokenizer, text: l
 
 
 # Function to calculate the steering vector``
-def calculate_steering_vector(model: AutoModelForCausalLM, tokenizer: AutoTokenizer, text_a, text_b, batch_size=16, device='cpu') -> list[torch.Tensor]:
+def calculate_steering_vector(
+    model: AutoModelForCausalLM,
+    tokenizer: AutoTokenizer,
+    text_a,
+    text_b,
+    batch_size=16,
+    device="cpu",
+) -> list[torch.Tensor]:
     means_a = compute_means(model, tokenizer, text_a, batch_size, device)
     means_b = compute_means(model, tokenizer, text_b, batch_size, device)
 
     steering_vector = [a - b for a, b in zip(means_a, means_b)]
     return steering_vector
+
+class Output(BaseModel):
+    output: bool
+def judge_output_llm(output):
+    load_dotenv()
+    openai_api_key = os.getenv("OPENAI_API_KEY")
+    if not openai_api_key:
+        raise ValueError("OPENAI_API_KEY is not set in the environment variables.")
+    openai = OpenAI(api_key=openai_api_key)
+    response = openai.chat.completions.parse(
+        model="gpt-3.5-turbo",
+        messages=[
+            {
+                "role": "system",
+                "content": "You are a judge that evaluates whether if the answer according to the question is opinionated or neutral , if opinionated then return 'True' else if neutral return 'False'",
+            },
+            {"role": "user", "content": f"{output}"},
+        ],
+        response_format=Output.model_json_schema(),
```

Comment:
- looks like good first thing to try! nice and straightforward
- nice idea with pydantic class to enforce a particular output format.